### ISSUE TITLE
[Released version] Ongoing timer resets when a notification is received

### ISSUE THREAD

**Issue by arquicanedo** [arquicanedo/barbecueboss#33]
I was grilling today and set a custom timer of 15:00. I was 3 minutes into the counter when I got a notification from WhatsApp. After the notification clears from the watch, the screen resets the timer.

I suspect this is also a bug on the [development] version. Perhaps with one of the methods of the BehaviorDelegate. 

I will debug this from the source code and on the watch. The simulator does not cover the notifications.

**Comment by nilsbenson** [comment]
We'll have to see what the lifecycle is for the app. I read somewhere that based on some things (memory pressure usually) the device will put the app to sleep sometimes and when it does that, the whole initialization happens again when it's brought back to the foreground.

I'm not sure if this means our timers are killed and restarted, or if we just need to avoid resetting things to zero based on a "we're running" flag we store in the apps settings?

**Comment by nilsbenson** [comment]
Is this the first time you've seen it or do you think this is a new issue? 

We're using about 55k in the simulator which is pretty close to the 64k max on the FR235. We could comment out for now the extra strings and pickers that we're not using to reduce that.

**Comment by arquicanedo** [comment]
No, this is the second time but I could not narrow down why it happened. I though I had clicked by accident but yesterday I confirmed that it is because I got the notification and screen changed. Finally the weekend is here so I'll debug the current development branch on the watch and test for that. If this behavior is gone with the simplified state machine then we don't have anything to do. May have been something related to the old state machine that was more intricate. The current way is very clean.
